File: 2024/March/_16/C/exerc3_soma.c
#include <stdio.h>

int soma(int v[], int total);
int somaRec(int v[], int pos, int total);

int soma(int v[], int total)
{
    return somaRec(v, 0, total);
}

int somaRec(int v[], int pos, int total)
{
    if (pos == total)
        return 0;
    return v[pos] + somaRec(v, pos+1, total);
}

int main()
{
    int vet[] = { 1, 2, 3, 4, 5, 6, 7, 8 };
    for (int i = 0; i < 8; i++)
        printf("%d ", vet[i]);
    printf("\n");
    printf("Soma: %d\n", soma(vet,8));
}

File: 2024/March/_16/Exerc3_Soma.java
package March._16;
public class Exerc3_Soma {

    public static int soma(int[] v)
    {
        return somaRec(v, 0);
    }

    public static int somaRec(int[] v, int pos)
    {
        if(pos == v.length)
            return 0;
        return v[pos] + somaRec(v, pos+1);
    }

    public static void main(String[] args) {

        int vet[] = { 1, 2, 3, 4, 5, 6, 7, 8 };
        for (int i: vet)
            System.out.printf("%d ", i);
        System.out.println();
        System.out.println("Soma: "+soma(vet));

    }
}
